### PR TITLE
Support new filter params

### PR DESCRIPTION
**Pull request by ryanmitchell**
Allow filtering by order_types (eg ?order_types=delivery,collection)
If delivery, also search delivery areas

See also: https://github.com/tastyigniter/TastyIgniter/pull/853

### CODE REVIEW

**Comment by sampoyigi** [comment]
Added the UI for order types filter and improved the sorting UI... Let me know what you think

**Comment by ryanmitchell** [comment]
Hmm not working for me until I visit a location page as I get an error on getOrderTypes() as there is no location selected. Do we have a way of knowing what order types are registered across all locations?

Is there a PR on orange theme for the locations page - I'm missing sorting options at the moment?

**Comment by sampoyigi** [comment]
Here's the theme changes https://github.com/tastyigniter/ti-theme-orange/pull/29

For location order types we use `options[offer_delivery]` and `$location->availableOrderTypes()->...->isDisabled()`

**Comment by ryanmitchell** [comment]
Yep sorry I wasnt clear enough... when you to to /locations without first going to a restaurant, you get an error as there is no location in your session.

```Call to a member function getOrderTypeOptions() on null```

the line is

```return Location::current()->getOrderTypeOptions();```

I've updated the code to show the approach needed.

Sorting tabs look good.

File: components/LocalList.php [hunk 1]
```diff
@@ -168,7 +168,14 @@ protected function getOrderType()
...
     {
-        return Location::current()->getOrderTypeOptions();
+        $types = [];
+        foreach (Locations_model::isEnabled()->get() as $location) {
+            foreach ($location->getOrderTypeOptions() as $locationOrderType) {
+                if (!in_array($locationOrderType, $types))
+                    $types[] = $locationOrderType;
+            }
+        }
+        return $types;
```
Comment (sampoyigi) [inline components/LocalList.php:178]: Not sure this is necessary as `Location::current()->getOrderTypeOptions()` returns all the registered order types and not just the location's order types..

Comment (sampoyigi) [reply components/LocalList.php:178]: Oops, I should start reading comments before reviewing!

We can use `OrderTypes::instance()->listOrderTypes()` to list all registered order types..

Comment (ryanmitchell) [reply components/LocalList.php:178]: Location::current() isnt always present... in that case maybe its use Locations_model::first()->getOrderTypeOptions() ?

**Pull request merged**